axum/src/docs/routing/nest.md

```diff
@@ -103,18 +73,49 @@ the prefix stripped:
 ```rust
 use axum::{routing::get, http::Uri, Router};
 
+let nested_router = Router::new()
+    .route("/", get(|uri: Uri| async {
+        // `uri` will _not_ contain `/bar`
+    }));
+
 let app = Router::new()
     .route("/foo/*rest", get(|uri: Uri| async {
         // `uri` will contain `/foo`
     }))
-    .nest("/bar", get(|uri: Uri| async {
-        // `uri` will _not_ contain `/bar`
-    }));
+    .nest("/bar", nested_router);
 # async {
 # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
 # };
 ```
 
+# Differences between `nest` and `nest_service`
+
+When [fallbacks] are called differs between `nest` and `nested_service`. Routers
+nested with `nest` will delegate to the fallback if they don't have a matching
+route, whereas `nested_service` will not.
+
+```rust
+use axum::{Router, routing::{get, any}, handler::Handler};
+
+let nested_router = Router::new().route("/users", get(|| async {}));
+
+let nested_service = Router::new().route("/app.js", get(|| async {}));
+
+async fn fallback() {}
+
+let app = Router::new()
+    .nest("/api", nested_router)
+    .nest_service("/assets", nested_service)
+    // the fallback is not called for request starting with `/bar` but will be
+    // called for requests starting with `/foo` if `nested_router` doesn't have
+    // a matching route
+    .fallback(fallback.into_service());
+# let _: Router = app;
+```
+
+Note that you would normally use [`tower_http::services::ServeDir`] for serving
+static files and thus not calling `nest_service` with a `Router`.
```

Comment:
```suggestion
static files and thus not call `nest_service` with a `Router`.
```
This fixes the grammar, although I'm still not actually clear on what this sentence wants to tell me 😅

Reply:
The example makes a router that with a `/app.js` route. So this sentence is mostly so people don't assume that that is how you server static files.

In general I think the use cases for using `nest_service` with `Router`s are limited but thats what we wanna show here to point out how the fallback behavior is different.

Reply:
I've tweaked the example a bit and hopefully made things more clear.